homeassistant/components/nextdns/config_flow.py

```diff
@@ -107,24 +137,57 @@ async def async_step_reauth_confirm(
     ) -> ConfigFlowResult:
         """Dialog that informs the user that reauth is required."""
         errors: dict[str, str] = {}
+        entry = self._get_reauth_entry()
 
         if user_input is not None:
-            try:
-                await async_init_nextdns(self.hass, user_input[CONF_API_KEY])
-            except InvalidApiKeyError:
-                errors["base"] = "invalid_api_key"
-            except (ApiError, ClientConnectorError, RetryError, TimeoutError):
-                errors["base"] = "cannot_connect"
-            except Exception:
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
-            else:
+            result = await async_validate_new_api_key(
+                self.hass, user_input, entry.data[CONF_PROFILE_ID]
+            )
+            if result.get("base") == "profile_not_available":
+                return self.async_abort(reason="profile_not_available")
```

Comment:
Why is aborting better instead of simply informing the user in the form so they can fix it directly?